pkgs/games/shticker-book-unwritten/default.nix

```
@@ -0,0 +1,58 @@
+{ alsaLib, buildFHSUserEnv, fetchFromGitHub, lib, libX11, libXext, libglvnd, openssl, pkg-config, rustPlatform, stdenvNoCC }:
+let
+
+  pname = "shticker-book-unwritten";
+  version = "1.0.3";
+
+  shticker-book-unwritten-unwrapped = rustPlatform.buildRustPackage rec {
+    inherit pname version;
+
+    src = fetchFromGitHub {
+      owner = "JonathanHelianthicusDoe";
+      repo = "shticker_book_unwritten";
+      rev = "v${version}";
+      sha256 = "08lyxica0b0vvivybsvzigy2j7saar78mbz723y3g5hqrilfb5np";
+    };
+
+    cargoPatches = [ ./cargo-lock.patch ];
+    cargoSha256 = "1lnhdr8mri1ns9lxj6aks4vs2v4fvg7mcriwzwj78inpi1l0xqk5";
+
+    nativeBuildInputs = [ pkg-config ];
+
+    buildInputs = [ openssl ];
+  };
+
+  # FHS env is needed to run the TTREngine binary, which the launcher downloads.
+  env = buildFHSUserEnv {
+    name = "${pname}-env-${version}";
+    targetPkgs = _: [
+      alsaLib
+      libX11
+      libXext
+      libglvnd
+      shticker-book-unwritten-unwrapped
+    ];
+    runScript = "shticker_book_unwritten";
+  };
+
+in stdenvNoCC.mkDerivation rec {
+  inherit pname version;
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  # Link the wrapped binary from the earlier FHS env into the derivation output
+  installPhase = ''
+    mkdir -p $out/bin
+    ln -s ${env}/bin/${env.name} $out/bin/shticker_book_unwritten
+  '';
+
+  meta = with lib; {
+    description = "Minimal CLI launcher for the Toontown Rewritten MMORPG";
+    homepage = "https://github.com/JonathanHelianthicusDoe/shticker_book_unwritten";
+    license = licenses.gpl3Plus;
+    maintainers = [ maintainers.reedrw ];
+    platforms = platforms.linux;
+  };
+}
```

I think this should be possible
```suggestion
  in buildFHSUserEnv {
    name = "${pname}-fhs-${version}";
    targetPkgs = pkgs: with pkgs; [
      alsaLib
      libX11
      libXext
      libglvnd
      shticker-book-unwritten-unwrapped
    ];
    runScript = "shticker_book_unwritten";
  };

  meta = with lib; {
    description = "Minimal CLI launcher for the Toontown Rewritten MMORPG";
    homepage = "https://github.com/JonathanHelianthicusDoe/shticker_book_unwritten";
    license = licenses.gpl3Plus;
    maintainers = [ maintainers.reedrw ];
    platforms = platforms.linux;
  };
}
```

---

I like using `mkDerivation` because if I just use `buildFHSUserEnv` without it, the binary in the package will be whatever `name` was set to. In this case it will output `./result/bin/shticker-book-unwritten-fhs-1.0.3`, when it should be `./result/bin/shticker_book_unwritten` as per the output of the Rust build.

I suppose I could just set the `name` to what I want the name of the binary to be as [lutris](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/lutris/fhsenv.nix#L15) does, to use another example from nixpkgs, but I like the package having a version number at the end (ex: `shticker-book-unwritten-1.0.3` vs just `shticker_book_unwritten`).